ngraph/core/src/op/roi_align.cpp

```diff
@@ -190,4 +202,104 @@ namespace ngraph
+                                                    out->get_data_ptr<bfloat16>(),
+                                                    feature_maps->get_shape(),
+                                                    rois->get_shape(),
+                                                    batch_indices->get_shape(),
+                                                    out->get_shape(),
+                                                    pooled_height,
+                                                    pooled_width,
+                                                    sampling_ratio,
+                                                    spatial_scale,
+                                                    pooling_mode);
+            break;
+        }
+        case element::Type_t::f16:
+        {
+            runtime::reference::roi_align<float16>(feature_maps->get_data_ptr<float16>(),
+                                                   rois->get_data_ptr<float16>(),
+                                                   batch_indices_vec_scaled_up.data(),
+                                                   out->get_data_ptr<float16>(),
+                                                   feature_maps->get_shape(),
+                                                   rois->get_shape(),
+                                                   batch_indices->get_shape(),
+                                                   out->get_shape(),
+                                                   pooled_height,
+                                                   pooled_width,
+                                                   sampling_ratio,
+                                                   spatial_scale,
+                                                   pooling_mode);
+            break;
+        }
+        case element::Type_t::f32:
+        {
+            runtime::reference::roi_align<float>(feature_maps->get_data_ptr<float>(),
+                                                 rois->get_data_ptr<float>(),
+                                                 batch_indices_vec_scaled_up.data(),
+                                                 out->get_data_ptr<float>(),
+                                                 feature_maps->get_shape(),
+                                                 rois->get_shape(),
+                                                 batch_indices->get_shape(),
+                                                 out->get_shape(),
+                                                 pooled_height,
+                                                 pooled_width,
+                                                 sampling_ratio,
+                                                 spatial_scale,
+                                                 pooling_mode);
+            break;
+        }
+        case element::Type_t::f64:
+        {
+            runtime::reference::roi_align<double>(feature_maps->get_data_ptr<double>(),
```

Comment:
Do we need evaluate for `f64` data? IE plugins don't support such precision and we don't add it to evaluate methods in order to decrease the impact on binary size.

Reply:
I will remove it then, documentation said the op should support all supported floating point types, so the fact that this type is present in for example `element::Type_t `suggested to me that it should be supported.

Reply:
Ok thank you!